core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1778,4 +1800,36 @@ class ParserTest extends munit.ScalaCheckSuite {
       assertEquals(left.parse(str), right.parse(str))
     }
   }
+
+  property("oneOf(string1(s)*) success => stringIn1(s*) success") {
+    forAll { (ss0: List[String], toParse: String) =>
+      val ss = ss0.filterNot(_.isEmpty)
+      val oneOfs = Parser.oneOf1(ss.map(Parser.string1))
+      val stringIn = Parser.stringIn1(ss)
+      if (oneOfs.parse(toParse).isRight) assert(stringIn.parse(toParse).isRight)
+    }
+  }
+
+  property("stringIn1(List(s)) == string1(s)") {
+    forAll { (s: String) =>
+      if (s.nonEmpty)
+        assertEquals(Parser.stringIn1(List(s)), Parser.string1(s))
+    }
+  }
+
+  property("stringIn1(List(s, s)) == string1(s)") {
+    forAll { (s: String) =>
+      if (s.nonEmpty)
+        assertEquals(Parser.stringIn1(List(s, s)), Parser.string1(s))
+    }
+  }
+
+  property("string1(s) matches  => stringIn1(ss) matches if s in ss") {
+    forAll { (s: String, ss0: List[String], toParse: String) =>
+      val ss = ss0.filterNot(_.isEmpty)
+      val ss1 = Random.shuffle(s :: ss)
+      if (s.nonEmpty && Parser.string1(s).parse(toParse).isRight)
+        assert(Parser.stringIn1(ss1).parse(toParse).isRight)
+    }
+  }
```

Comment:
isn't it also true that:
```
P.string1(s0).orElse(P.string1(s1)).parse(arg).toOption ==
  P.stringIn1(s0 :: s1 :: Nil).parse(arg).toOption
```
That is: if we union to string parsers we get the same as a stringIn?

Also, doesn't that also hold for two stringIn parsers?
```
P.stringIn1(s0).orElse(P.stringIn1(s1)).parse(arg).toOption ==
  P.stringIn1(s0 ::: s1).parse(arg).toOption
```

Reply:
Okay, I don't think these laws are true in general. I think they are only true if the first item in the orElse has no prefixes in the second. I think we can check for that in the law:
```scala
val noPrefix = left.forall { s => !right.exists(_.startsWith(s)) }
```
something like that.

finally, here is one more law:
```scala
forAll { (ss0: List[String], toParse: String) =>
  val ss = ss0.filterNot(_.isEmpty)
  val left = P.stringIn1(ss).parse(toParse).toOption
  val right = ss.filter(toParse.startsWith(_)).sortBy { s => -s.length }
  assertEquals(left, right.headOption)
}
```

Reply:
If `s0` is a prefix of `s1` then result is different when `arg` matches `s1`. The thing is that the radix tree approach really doesn't care about the order of the alternatives, whereas the `orElse` approach does and takes the first one that matches, eagerly.

Reply:
Yes, that's why I suggested the modification above. Maybe our comments crossed in flight.

Reply:
Oh I didn't se that when replying, my bad. I will add these laws.